visualization/js/renderNetworkGraph.js

```diff
@@ -1310,3 +1312,17 @@ function toggleTutorial(button) {
         hidehints();
     }
 }
+
+// Programmatically add tooltips to the small info circles on the buttons on the sidebar.
+// See filter_data.js for dictionary of tooltip text descriptions.
+function addTooltipText() {
+    for (let key in tooltips_text) {
+        // console.log(key)
+        document.getElementById(key)
+            .setAttribute("data-toggle", "tooltip");
+        document.getElementById(key)
+            .setAttribute("data-placement", "top");
+        document.getElementById(key)
+            .setAttribute("title", tooltips_text[key]);
```

Comment:
I would hold the element found using `document.getElementById(key)` in a variable and then do a set attribute on it as you are doing right now, which prevents the 3 separate calls to document.getElementById. Otherwise looks great!